src/_tutorials/dart-vm/httpserver.md

```diff
@@ -555,66 +560,55 @@ The `close()` method sends the request to the server
+import 'dart:convert' show UTF8, JSON;
+
+Future main() async {
+  Map jsonData = {
+    'name': 'Han Solo',
+    'job': 'reluctant hero',
+    'BFF': 'Chewbacca',
+    'ship': 'Millennium Falcon',
+    'weakness': 'smuggling debts'
+  };
+
+  var request = await new HttpClient()
+      [!/*1*/!] .post(InternetAddress.LOOPBACK_IP_V4.host, 4049, 'file.txt');
+  [!/*2*/!] request.headers.contentType = ContentType.JSON;
+  [!/*3*/!] request.write(JSON.encode(jsonData));
+  [!/*4*/!] HttpClientResponse response = await request.close();
+  [!/*5*/!] await for (var contents in response.transform([!UTF8.decoder!])) {
+    print(contents);
   }
+}
 {% endprettify %}
 <div class="prettify-filename">basic_writer_client.dart</div>
 
-<span class="code-note">1</span>
-When the `post()` connection succeeds (and execution resumes after
-the await expression), the returned [HttpClientRequest][]
-object is assigned to the `request` variable.
-
-<span class="code-note">2</span>
-The `post()` method requires the host, port, and the path to the requested
-resource.
-In addition to `post()`, the HttpClient class provides functions
-for making other kinds of
-requests, including `postUrl()`, `get()`, and `open()`.
-
-<span class="code-note">3</span>
-An HttpClientRequest object has an HttpHeaders object,
-which contains the request headers.
-For some headers,
-like `contentType`,
-HttpHeaders has a property specific to that header.
-For other headers, use the `set()` method to
-put the header in the HttpHeaders object.
-
-<span class="code-note">4</span>
-The client writes data to the request object using `write()`.
-The encoding, JSON in this example,
-matches the type specified in the ContentType header.
-
-<span class="code-note">5</span>
-The `close()` method sends the request to the server and, when complete,
-returns an [HttpClientResponse][] object
-that's assigned to the `reponse` variable.
-
-<span class="code-note">6</span>
-The response from the server is encoded in UTF-8.
-Use a transformer defined in the `dart:convert` library
-to convert the data into regular Dart string format.
+{:.code-notes}
+1. The `post()` method requires the host, port, and the path to the requested
+   resource. In addition to `post()`, the [HttpClient][] class provides
+   functions for making other kinds of requests, including `postUrl()`,
+   `get()`, and `open()`.
+
+2. An [HttpClientRequest][] object has an [HttpHeaders][] object, which
+   contains the request headers. For some headers, like `contentType`,
+   HttpHeaders has a property specific to that header. For other headers, use
+   the `set()` method to put the header in the HttpHeaders object.
+
+3. The client writes data to the request object using `write()`. The encoding,
+   JSON in this example, matches the type specified in the [ContentType][]
+   header.
+
+4. The `close()` method sends the request to the server and, when complete,
+   returns an [HttpClientResponse][] object that's assigned to the `reponse`
```

Comment:
reponse -> response

(old typo!)

Reply:
Good catch! (This is another reason why I like to avoid mentioning variable names in the prose.)

Actually, I've dropped the last part of the sentence. It now reads:

> The `close()` method sends the request to the server and, when complete,  returns an [HttpClientResponse][] object.

The prose should be commenting on the essential parts of the code and skip uninteresting details.